sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScanBase.scala

```diff
@@ -871,6 +872,28 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     currentChunk
   }
 
+  /**
+   * Take case-sensitive into consideration when getting the data reading column names
+   * before sending parquet-format buffer to cudf.
+   *
+   * @param readDataSchema spark schema to read
+   * @param fileSchema  the schema of the dumped parquet-format buffer
+   * @param isCaseSensitive if it is case sensitive
+   * @return a sequence of column names following the order of readDataSchema
+   */
+  protected def toCudfColumnNames(
+      readDataSchema: StructType,
+      fileSchema: MessageType,
+      isCaseSensitive: Boolean): Seq[String] = {
+    val fields = fileSchema.getFields.asScala.map(_.getName).toSet
```

Comment:
Is this approach sufficient to handle nested struct field names, or will this only handle top-level column names?

Reply:
Yeah, since `ParquetReadSupport.clipParquetSchema` has helped us to clip out the unnecessary columns

Reply:
I'm not sure why clipping has anything to do with my question, as this isn't a problem of failing to remove columns and accidentally loading the wrong column but rather mapping Spark's requested names to the column names in the file.

Reply:
Sorry for my wrong answer. Yeah, we only handle the top-level columns.

Reply:
Is there a followup needed for this, or does the case-insensitivity only apply to top-level columns?

Reply:
Seems we don't need a follow up. Since ParquetReadSupport.clipParquetSchema will help us to filter the necessary columns according to the flag of case sensitive, which means all columns are needed in the clipped schema. So in case-insensitive scenario, matching the top-level columns should be ok.

Reply:
> Since ParquetReadSupport.clipParquetSchema will help us to filter the necessary columns according to the flag of case sensitive

We're filtering the columns based on case-insensitivity, but are we requesting them in that way as well?  Specifically I'm thinking of the nested pruning case, where we ask the reader to load individual fields of a struct column without loading the entire struct column.

For example, consider a struct column named x containing three fields: a, b, c.  Spark asks to load column X.A.  We will clip to just include x.a, but when we ask libcudf to load the column, will we also fix the column name to be x.a or will we ask for x.A?  The latter will probably fail since libcudf is case-sensitive in its column lookup.  We probably need a test case for nested pruning case insensitivity handling.

Reply:
Hi Jason, Yeah, we're only caring the top-level columns. For your example, we only pass the top-level "x" to libcudf.

Reply:
I just tried to specify the user defined schema which can repro this issue without involving any hive things. And the new commit has added the integration test.

Reply:
> Yeah, we're only caring the top-level columns.

For nested pruning that's not true, correct?  That's a case where we're not loading an entire top-level struct column but only individual fields from it.